regulations/static/regulations/js/source/views/comment/comment-view.js

```diff
@@ -171,6 +182,27 @@ var CommentView = Backbone.View.extend({
     });
     this.attachmentViews[index].remove();
     this.attachmentViews.splice(index, 1);
+    this.setAttachmentCount();
+  },
+
+  setAttachmentCount: function() {
+    // Sum saved attachments on other comments and pending attachments on the
+    // current comment
+    var saved = comments.filter(this.options.docId).map(function(comment) {
+      return comment.id !== this.model.id ?
+        comment.get('files').map(function(file) {
+          return file.key;
+        }) :
+        [];
+    }.bind(this));
+    var pending = this.attachmentViews.map(function(view) {
+      return view.options.key;
+    });
+    var keys = _.flatten([].concat(saved, pending));
+    this.attachmentCount = keys.length;
```

Comment:
While I'm generally a fan of the idioms, the nested `map`s and `flatten` seem like overkill here. Would a counter and calls to `.length` be easier to read?